File: C_C++/LuoGu/1-3 Enumeration/14 P2036.cpp
#include <iostream>
#include <limits.h>
using namespace std;

int main(int argc, char *argv[]) {

    int n;
    cin >> n;
    long long s[10];
    long long b[10];
    for(int i = 0; i < n; i++)
        cin >> s[i] >> b[i];

    long long ans = LONG_LONG_MAX;
    for(int state = 1; state < (1 << n); state++) {
        long long mul = 1;
        long long sum = 0;
        for(int i = 0; i < n; i++)
            if((1 << i) & state) {
                mul *= s[i];
                sum += b[i];
            }
        ans = min(abs(mul - sum), ans);
    }

    cout << ans;

    return 0;
}

File: C_C++/LuoGu/1-3 Enumeration/14 P2036.lua
local function main()
    local n = io.read("*n");
    local s = {};
    local b = {};

    for i = 1, n do
        s[i] = io.read("*n");
        b[i] = io.read("*n");
    end

    local min_diff = math.huge;

    for state = 1, (1 << n) - 1 do
        local mul = 1;
        local sum = 0;
        local temp = state

        for i = 1, n do
            local bit = temp % 2;
            temp = math.floor(temp / 2)
            if bit == 1 then
                mul = mul * s[i];
                sum = sum + b[i];
            end
        end

        local diff = math.abs(mul - sum);
        if diff < min_diff then
            min_diff = diff
        end
    end
    print(min_diff)
end

main();